pkg/cmd/grafana-cli/commands/install_command_test.go

```diff
@@ -37,3 +39,39 @@ func TestFoldernameReplacement(t *testing.T) {
 		})
 	})
 }
+
+func TestExtractFiles(t *testing.T) {
+	Convey("Should preserve file permissions for plugin backend binaries for linux and darwin", t, func() {
+		err := os.RemoveAll("./testfiles/fake-plugins-dir/grafana-simple-json-datasource")
+		So(err, ShouldBeNil)
+
+		body, err := ioutil.ReadFile("./testfiles/grafana-simple-json-datasource-ec18fa4da8096a952608a7e4c7782b4260b41bcf.zip")
+		So(err, ShouldBeNil)
+
+		err = extractFiles(body, "grafana-simple-json-datasource", "./testfiles/fake-plugins-dir")
+		So(err, ShouldBeNil)
+
+		//File in zip has permissions 777
```

Comment:
Comments are wrong. 755 not 777

Reply:
I would probably go with: `Binary should be +rx to everyone.`

Reply:
The original file is 777 so testing that it gets set to 755. The scenario is that when publishing plugins the permissions could be incorrect.